[SIMULATION] [GCC12] Fix build warnings

smuzaffar opened the pull request:
This should fix the build warnings for gcc12 IBs
```
In copy constructor 'HFShower::Hit::Hit(const HFShower::Hit&)',
    inlined from 'void std::__new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = HFShower::Hit; _Args = {const HFShower::Hit&}; _Tp = HFShower::Hit]' at ...external/gcc/12.2.0-f8ec77b592790702d83afb7106a458e3/include/c++/12.2.1/bits/new_allocator.h:175:4,
    inlined from 'static void std::allocator_traits<std::allocator<_CharT> >::construct(allocator_type&, _Up*, _Args&& ...) [with _Up = HFShower::Hit; _Args = {const HFShower::Hit&}; _Tp = HFShower::Hit]' at ...external/gcc/12.2.0-f8ec77b592790702d83afb7106a458e3/include/c++/12.2.1/bits/alloc_traits.h:516:17,
    inlined from 'void std::vector<_Tp, _Alloc>::push_back(const value_type&) [with _Tp = HFShower::Hit; _Alloc = std::allocator<HFShower::Hit>]' at ...external/gcc/12.2.0-f8ec77b592790702d83afb7106a458e3/include/c++/12.2.1/bits/stl_vector.h:1281:30,
    inlined from 'std::vector<HFShower::Hit> HFShower::getHits(const G4Step*, bool)' at ...cms/cmssw/CMSSW_12_6_X_2022-09-24-1100/src/SimG4CMS/Calo/src/HFShower.cc:348:19:
  ...cms/cmssw/CMSSW_12_6_X_2022-09-24-1100/src/SimG4CMS/Calo/interface/HFShower.h:32:10: warning: 'hit.HFShower::Hit::depth' may be used uninitialized [-Wmaybe-uninitialized]
    32 |   struct Hit {
      |          ^~~
...cms/cmssw/CMSSW_12_6_X_2022-09-24-1100/src/SimG4CMS/Calo/src/HFShower.cc: In member function 'std::vector<HFShower::Hit> HFShower::getHits(const G4Step*, bool)':
...cms/cmssw/CMSSW_12_6_X_2022-09-24-1100/src/SimG4CMS/Calo/src/HFShower.cc:280:17: note: 'hit' declared here
  280 |   HFShower::Hit hit;
------
CMSSW_12_6_X_2022-09-24-1100/src/SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc: In constructor 'SiPixelChargeReweightingAlgorithm::SiPixelChargeReweightingAlgorithm(const edm::ParameterSet&, edm::ConsumesCollector)':
  CMSSW_12_6_X_2022-09-24-1100/src/SimTracker/SiPixelDigitizer/plugins/SiPixelChargeReweightingAlgorithm.cc:65:15: warning: member 'SiPixelChargeReweightingAlgorithm::templateStores_' is used uninitialized [-Wuninitialized]
    65 |       templ2D(templateStores_),
```

smuzaffar committed:
[SIMULATION] [GCC12] Fix build warnings

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39523/32287

- This PR adds an extra 16KB to repository

cmsbuild commented:
A new Pull Request was created by  @smuzaffar (Malik Shahzad Muzaffar) for master.

It involves the following packages:

- SimG4CMS/Calo (**simulation**)
- SimTracker/SiPixelDigitizer (**simulation**)


@cmsbuild, @civanch, @mdhildreth can you please review it and eventually sign? Thanks.
@tvami, @fabiocos, @VourMa, @missirol, @felicepantaleo, @GiacomoSguazzoni, @JanFSchulte, @rovere, @VinInn, @bsunanda, @OzAmram, @thomreis, @ferencek, @mtosi, @mmusich, @dkotlins, @threus, @dgulhan, @slomeo, @simonepigazzini this is something you requested to watch as well.
@perrotta, @dpiparo, @rappoccio you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

smuzaffar commented:
please test

civanch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

perrotta commented on SimG4CMS/Calo/src/HFShower.cc:
```diff
@@ -341,6 +341,7 @@ std::vector<HFShower::Hit> HFShower::getHits(const G4Step *aStep, bool forLibrar
   std::vector<double> momz = cherenkov_->getMom();
 
   for (int i = 0; i < npe; ++i) {
+    hit.depth = 0;
```
I would have (naively) thought that here the `depth` calculated above should have been assigned instead of 0...
Just fpr check: is `hit.depth = 0` the correct assignment here?

smuzaffar replied:
you are right @perrotta , at https://github.com/cms-sw/cmssw/blob/99e00f24e2a0423766f85da9b55d469c1f35a2b2/SimG4CMS/Calo/src/HFShower.cc#L240 also does the same. I have updated it to use the depth calculated above

smuzaffar committed:
properly initialize hit depth

same as it is done at https://github.com/cms-sw/cmssw/blob/99e00f24e2a0423766f85da9b55d469c1f35a2b2/SimG4CMS/Calo/src/HFShower.cc#L240

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39523/32304

- This PR adds an extra 16KB to repository

perrotta commented:
please test

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @rappoccio (and backports should be raised in the release meeting by the corresponding L2)

perrotta commented:
+1
